daemon/api.go

```diff
@@ -254,46 +254,67 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 	}
 	defer lock.Unlock()
 
-	sources := make([]string, 1, 2)
-	sources[0] = "local"
-	// we're not worried if the remote repos error out
-	found, _ := newRemoteRepo().All()
-	if len(found) > 0 {
-		sources = append(sources, "store")
+	results := make(map[string]map[string]interface{})
+	sources := make([]string, 0, 2)
+	query := r.URL.Query()
+
+	var includeStore, includeLocal bool
+	if len(query["sources"]) > 0 {
+		for _, v := range strings.Split(query["sources"][0], ",") {
+			if v == "store" {
+				includeStore = true
+			} else if v == "local" {
+				includeLocal = true
+			}
+		}
+	} else {
+		includeStore = true
+		includeLocal = true
```

Comment:
Hmm, but it also means we need further code inside the first branch. Nevermind, sorry, your alternative is just as good.